### PR TITLE
Doxygen-ignore cpp files

### LINKED ISSUE #3050: Should we parse only headers with Doxygen?
We currently parse both headers and source files with Doxygen, but exclude a lot of template-related problematic code snippets in source files by wrapping them in `/// \cond`. Most (all?) of our documentation is in header files, so how about we exclude sources files from Doxygen altogether?

### PR DESCRIPTION
**Pull request by nilsvu**
## Proposed changes

- All our documentation is in header files, so this PR just removes cpp files from the Doxygen input altogether. Parsing cpp files with Doxygen has been a nuisance because Doxygen has trouble with some template expressions. The change means we don't have to `/// \cond` problematic code in cpp files anymore (though we still have to do it in header files).
- Also adds a file check that cpp files contain no Doxygen formatting. We have been checking this in code review so far.
- Deletes all Doxygen-formatting from cpp files (including the `/// \cond`s) so the file check passes. We could leave the `/// \cond`s in the code (it's a separate commit so I can easily drop it), but then we can't add the file check.

Closes #3050.

### Upgrade instructions

- You don't have to Doxygen-ignore problematic code in cpp files anymore using `/// \cond`-comments (though you may still have to do so in header files).
- For Doxygen `\snippet`s write `// [snippet_name]`, not `/// [snippet_name]`.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `major new feature` if appropriate.

### CODE REVIEW

**Comment by nilsdeppe** [comment]
have you verified that our snippets show up correctly? Those are the only things in cpp files

**Comment by nilsvu** [comment]
I did, seems to work fine. You can double-check by downloading the built docs from the CI run.

File: tests/Unit/Evolution/DiscontinuousGalerkin/Actions/Test_ApplyBoundaryCorrections.cpp [hunk 1]
```diff
@@ -81,12 +81,10 @@ struct BoundaryTerms final : public BoundaryCorrection<Dim> {
     using type = Scalar<DataVector>;
   };
 
-  /// \cond
-  explicit BoundaryTerms(CkMigrateMessage* /*unused*/) noexcept {}
+    explicit BoundaryTerms(CkMigrateMessage* /*unused*/) noexcept {}
```
Comment (nilsdeppe) [inline tests/Unit/Evolution/DiscontinuousGalerkin/Actions/Test_ApplyBoundaryCorrections.cpp]: why is the indentation changing in some of these files?

**Review by nilsdeppe — CHANGES_REQUESTED** [review]

File: tests/Unit/Parallel/Test_GlobalCache.cpp [hunk 2]
```diff
@@ -219,11 +219,9 @@ struct TestMetavariables {
 // PerformAlgorithmCallback because they can be mocked.
 class UseCkCallbackAsCallback : public Parallel::Callback {
  public:
-  /// \cond
-  explicit UseCkCallbackAsCallback(CkMigrateMessage* /*unused*/) noexcept {}
+    explicit UseCkCallbackAsCallback(CkMigrateMessage* /*unused*/) noexcept {}
   WRAPPED_PUPable_decl(UseCkCallbackAsCallback);
-  /// \endcond
-  explicit UseCkCallbackAsCallback(const CkCallback& callback)
+    explicit UseCkCallbackAsCallback(const CkCallback& callback)
```
Comment (kidder) [inline tests/Unit/Parallel/Test_GlobalCache.cpp]: fix indentation

File: tests/Unit/Parallel/Test_GlobalCache.cpp [hunk 3]
```diff
@@ -219,11 +219,9 @@ struct TestMetavariables {
 // PerformAlgorithmCallback because they can be mocked.
 class UseCkCallbackAsCallback : public Parallel::Callback {
  public:
-  /// \cond
-  explicit UseCkCallbackAsCallback(CkMigrateMessage* /*unused*/) noexcept {}
+    explicit UseCkCallbackAsCallback(CkMigrateMessage* /*unused*/) noexcept {}
```
Comment (kidder) [inline tests/Unit/Parallel/Test_GlobalCache.cpp]: fix indentation

File: tests/Unit/Parallel/Test_AlgorithmPhaseControl.cpp [hunk 4]
```diff
@@ -104,12 +102,10 @@ struct TempPhaseATrigger : public Trigger<TriggerRegistrars> {
 template <typename TriggerRegistrars =
               tmpl::list<Registrars::TempPhaseBTrigger>>
 struct TempPhaseBTrigger : public Trigger<TriggerRegistrars> {
-  /// \cond
-  TempPhaseBTrigger() = default;
+    TempPhaseBTrigger() = default;
```
Comment (kidder) [inline tests/Unit/Parallel/Test_AlgorithmPhaseControl.cpp]: fix indentation

File: tests/Unit/Parallel/Test_AlgorithmPhaseControl.cpp [hunk 5]
```diff
@@ -83,12 +83,10 @@ struct ComponentBeta;
 template <typename TriggerRegistrars =
               tmpl::list<Registrars::TempPhaseATrigger>>
 struct TempPhaseATrigger : public Trigger<TriggerRegistrars> {
-  /// \cond
-  TempPhaseATrigger() = default;
+    TempPhaseATrigger() = default;
```
Comment (kidder) [inline tests/Unit/Parallel/Test_AlgorithmPhaseControl.cpp]: fix indentation

File: tests/Unit/Parallel/PhaseControl/Test_PhaseChange.cpp [hunk 6]
```diff
@@ -73,12 +73,10 @@ bool contributed = false;
 template <typename PhaseChangeRegistrars =
               tmpl::list<Registrars::TestPhaseChange>>
 struct TestPhaseChange : public PhaseChange<PhaseChangeRegistrars> {
-  /// \cond
-  TestPhaseChange() = default;
+    TestPhaseChange() = default;
```
Comment (kidder) [inline tests/Unit/Parallel/PhaseControl/Test_PhaseChange.cpp]: fix indentation

File: tests/Unit/Parallel/PhaseControl/Test_ExecutePhaseChange.cpp [hunk 7]
```diff
@@ -38,13 +38,11 @@ template <PhaseControl::ArbitrationStrategy Strategy, size_t Index,
...
 struct TestPhaseChange : public PhaseChange<PhaseChangeRegistrars> {
-  /// \cond
-  TestPhaseChange() = default;
+    TestPhaseChange() = default;
   explicit TestPhaseChange(CkMigrateMessage* /*unused*/) noexcept {}
   using PUP::able::register_constructor;
   WRAPPED_PUPable_decl_template(TestPhaseChange);  // NOLINT
-  /// \endcond
-  static std::string name() noexcept {
+    static std::string name() noexcept {
```
Comment (kidder) [inline tests/Unit/Parallel/PhaseControl/Test_ExecutePhaseChange.cpp]: fix indentation

File: tests/Unit/Parallel/PhaseControl/Test_ExecutePhaseChange.cpp [hunk 8]
```diff
@@ -38,13 +38,11 @@ template <PhaseControl::ArbitrationStrategy Strategy, size_t Index,
           typename PhaseChangeRegistrars =
               tmpl::list<Registrars::TestPhaseChange<Strategy, Index>>>
 struct TestPhaseChange : public PhaseChange<PhaseChangeRegistrars> {
-  /// \cond
-  TestPhaseChange() = default;
+    TestPhaseChange() = default;
```
Comment (kidder) [inline tests/Unit/Parallel/PhaseControl/Test_ExecutePhaseChange.cpp]: fix indentation

File: tests/Unit/IO/Observers/Test_RegisterEvents.cpp [hunk 9]
```diff
@@ -47,11 +47,9 @@ class SomeEvent : public Event<EventRegistrars> {
   explicit SomeEvent(std::string subfile_path) noexcept
       : subfile_path_(std::move(subfile_path)) {}
 
-  /// \cond
-  explicit SomeEvent(CkMigrateMessage* /*unused*/) noexcept {}
+    explicit SomeEvent(CkMigrateMessage* /*unused*/) noexcept {}
```
Comment (kidder) [inline tests/Unit/IO/Observers/Test_RegisterEvents.cpp]: fix indentation

File: tests/Unit/Evolution/DiscontinuousGalerkin/Test_BoundaryCorrectionsHelper.cpp [hunk 10]
```diff
@@ -141,12 +139,10 @@ struct Correction final : public CorrectionBase {
...
   }
 
-  /// \cond
-  explicit Correction(CkMigrateMessage* msg) noexcept : CorrectionBase(msg) {}
+    explicit Correction(CkMigrateMessage* msg) noexcept : CorrectionBase(msg) {}
   using PUP::able::register_constructor;
   WRAPPED_PUPable_decl_template(Correction);  // NOLINT
-  /// \endcond
-  void pup(PUP::er& p) override { CorrectionBase::pup(p); }
+    void pup(PUP::er& p) override { CorrectionBase::pup(p); }
```
Comment (kidder) [inline tests/Unit/Evolution/DiscontinuousGalerkin/Test_BoundaryCorrectionsHelper.cpp]: fix indentation

File: tests/Unit/Evolution/DiscontinuousGalerkin/Test_BoundaryCorrectionsHelper.cpp [hunk 11]
```diff
@@ -141,12 +139,10 @@ struct Correction final : public CorrectionBase {
     return std::make_unique<Correction>(*this);
   }
 
-  /// \cond
-  explicit Correction(CkMigrateMessage* msg) noexcept : CorrectionBase(msg) {}
+    explicit Correction(CkMigrateMessage* msg) noexcept : CorrectionBase(msg) {}
```
Comment (kidder) [inline tests/Unit/Evolution/DiscontinuousGalerkin/Test_BoundaryCorrectionsHelper.cpp]: fix indentation

File: tests/Unit/Evolution/DiscontinuousGalerkin/Test_BoundaryCorrectionsHelper.cpp [hunk 12]
```diff
@@ -105,12 +105,10 @@ struct CorrectionBase : public PUP::able {
 
   explicit CorrectionBase(CkMigrateMessage* msg) noexcept : PUP::able(msg) {}
 
-  /// \cond
-#pragma GCC diagnostic push
+  #pragma GCC diagnostic push
```
Comment (kidder) [inline tests/Unit/Evolution/DiscontinuousGalerkin/Test_BoundaryCorrectionsHelper.cpp]: fix indentation

File: tests/Unit/Evolution/DiscontinuousGalerkin/Actions/Test_BoundaryConditions.cpp [hunk 13]
```diff
@@ -127,12 +127,10 @@ struct BoundaryTerms final
...
   };
 
-  /// \cond
-  explicit BoundaryTerms(CkMigrateMessage* /*unused*/) noexcept {}
+    explicit BoundaryTerms(CkMigrateMessage* /*unused*/) noexcept {}
   using PUP::able::register_constructor;
   WRAPPED_PUPable_decl_template(BoundaryTerms);  // NOLINT
-  /// \endcond
-  BoundaryTerms(const bool mesh_is_moving, const double sign_of_normal)
+    BoundaryTerms(const bool mesh_is_moving, const double sign_of_normal)
```
Comment (kidder) [inline tests/Unit/Evolution/DiscontinuousGalerkin/Actions/Test_BoundaryConditions.cpp]: fix indentation

File: tests/Unit/Evolution/DiscontinuousGalerkin/Actions/Test_BoundaryConditions.cpp [hunk 14]
```diff
@@ -127,12 +127,10 @@ struct BoundaryTerms final
     using type = Scalar<DataVector>;
   };
 
-  /// \cond
-  explicit BoundaryTerms(CkMigrateMessage* /*unused*/) noexcept {}
+    explicit BoundaryTerms(CkMigrateMessage* /*unused*/) noexcept {}
```
Comment (kidder) [inline tests/Unit/Evolution/DiscontinuousGalerkin/Actions/Test_BoundaryConditions.cpp]: fix indentation

File: tests/Unit/Evolution/DiscontinuousGalerkin/Actions/Test_ApplyBoundaryCorrections.cpp [hunk 15]
```diff
@@ -81,12 +81,10 @@ struct BoundaryTerms final : public BoundaryCorrection<Dim> {
...
   using PUP::able::register_constructor;
   WRAPPED_PUPable_decl_template(BoundaryTerms);  // NOLINT
-  /// \endcond
-  BoundaryTerms() = default;
+    BoundaryTerms() = default;
```
Comment (kidder) [inline tests/Unit/Evolution/DiscontinuousGalerkin/Actions/Test_ApplyBoundaryCorrections.cpp]: fix indentation

File: tests/Unit/Evolution/DiscontinuousGalerkin/Actions/Test_ApplyBoundaryCorrections.cpp (hunk 1 above)
Comment (kidder) [reply tests/Unit/Evolution/DiscontinuousGalerkin/Actions/Test_ApplyBoundaryCorrections.cpp]: please fix the indentation changes in a separate fixup commit to make it easier to check that you did them

**Review by kidder — CHANGES_REQUESTED** [review]

File: tests/Unit/Evolution/DiscontinuousGalerkin/Actions/Test_ApplyBoundaryCorrections.cpp (hunk 1 above)
Comment (nilsvu) [reply tests/Unit/Evolution/DiscontinuousGalerkin/Actions/Test_ApplyBoundaryCorrections.cpp]: Good catch! In the regex-replace I didn't include the whitespace preceding the `/// \cond`, so it got added to the following line. I repeated the regex-replace and pushed the diff as a fixup.

**Review by nilsdeppe — COMMENTED** [review]
I'm fine with you squashing once @kidder has taken a look

**Review by kidder — COMMENTED** [review]
you can squash

**Review by kidder — DISMISSED** [review]

**Comment by nilsvu** [comment]
Rebased and fixed a bunch of trivial clang-tidy issues

**Review by nilsdeppe — DISMISSED** [review]

**Comment by nilsvu** [comment]
Rebased again

**Review by nilsdeppe — APPROVED** [review]

**Review by kidder — APPROVED** [review]

**Pull request merged**